src/AWS/CRT/HTTP/Headers.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace AWS\CRT\HTTP;
+
+use AWS\CRT\Internal\Encoding;
+
+final class Headers {
+    private $headers = [];
+
+    public function __construct($headers = []) {
+        $this->headers = $headers;
+    }
+
+    public static function marshall($headers) {
+        $buf = "";
+        foreach ($headers->headers as $header => $value) {
+            $buf .= Encoding::encodeString($header);
```

Comment:
Trying to understand this-- how would you know where each header begins and ends if you're just encoding all of it into a long byte array?  Just because you're padding the string so you know how long each header is?  If so, how will this handle longer headers?

Reply:
it's encoded as length|string|length|string|...etc. We use this format in other CRTs to pack HTTP requests between languages.